aiohttp/payload.py

```diff
@@ -307,17 +307,36 @@ def __init__(
             if hdrs.CONTENT_DISPOSITION not in self.headers:
                 self.set_content_disposition(disposition, filename=self._filename)
 
+        self._writable = True
+
+        try:
+            self._seekable = self._value.seekable()
+        except AttributeError:  # https://github.com/python/cpython/issues/124293
+            self._seekable = False
+
+        if self._seekable:
+            self._stream_pos = self._value.tell()
+        else:
+            self._stream_pos = 0
+
     async def write(self, writer: AbstractStreamWriter) -> None:
         loop = asyncio.get_event_loop()
-        try:
+        if self._seekable:
```

Comment:
This can be combined into a single executor job. Example

```diff
diff --git a/aiohttp/payload.py b/aiohttp/payload.py
index 395c44f6e..3b222b93e 100644
--- a/aiohttp/payload.py
+++ b/aiohttp/payload.py
@@ -319,15 +319,19 @@ class IOBasePayload(Payload):
         else:
             self._stream_pos = 0
 
-    async def write(self, writer: AbstractStreamWriter) -> None:
-        loop = asyncio.get_event_loop()
+    def _read_first(self) -> None:
+        """Read the first chunk of data from the stream."""
         if self._seekable:
-            await loop.run_in_executor(None, self._value.seek, self._stream_pos)
+            self._value.seek(self._stream_pos)
         elif not self._writable:
             raise RuntimeError(
                 f'Non-seekable IO payload "{self._value}" is already consumed (possibly due to redirect, consider storing in a seekable IO buffer instead)'
             )
-        chunk = await loop.run_in_executor(None, self._value.read, 2**16)
+        return self._value.read(2**16)
+
+    async def write(self, writer: AbstractStreamWriter) -> None:
+        loop = asyncio.get_running_loop()
+        chunk = await loop.run_in_executor(None, self._read_first)
         while chunk:
             await writer.write(chunk)
             chunk = await loop.run_in_executor(None, self._value.read, 2**16)
```